File: recursion/java/Recursion3.java
import java.util.Scanner;

public class Recursion3 {

    public static void printFirstNEvenNaturalNumber(int n) {
        // negative number
        if (n <= 1) {

            return;
        }

        if (n == 2) {
            System.out.print(n + ", ");
            return;
        }
        // make even
        if ((n & 1) >= 1) {
            n = n - 1;
        }

        printFirstNEvenNaturalNumber(n - 2);
        System.out.print(n + ", ");

    }

    public static Scanner scan = new Scanner(System.in);

    public static void main(String... args) {
        int n = scan.nextInt();
        printFirstNEvenNaturalNumber(n);
    }
}

File: recursion/go/Recursion3.go
package main

import "fmt"

// first n natural number
func printFirstNEvenNaturalNumber(number int) {
	// negative number
	if number <= 1 {
		return
	}

	// base case
	if number == 2 {
		fmt.Print(2, ", ")
		return
	}

	// start with even 
if(number & 1 >= 1){
	 number = number -1
}
	// recursive call
	printFirstNEvenNaturalNumber(number - 2)
	fmt.Print(number, ", ")

}
func main() {
	var number int
	fmt.Scan(&number)
	printFirstNEvenNaturalNumber(number)
}
